tests/integration/test_script_runner.py

```diff
@@ -660,55 +759,35 @@ def _execute_integration_test(test_configuration, tmp_path):
                 shutil.copyfile(src=source_file, dst=dest_file)
 
         # UAT Script
+        user_flow_script = integration_test_fixture.user_flow_script
         script_source = os.path.join(
             base_dir,
-            test_configuration.get("user_flow_script"),
+            user_flow_script,
         )
         script_path = os.path.join(tmp_path, "test_script.py")
         shutil.copyfile(script_source, script_path)
 
-        # Util Script
-        if test_configuration.get("util_script") is not None:
-            script_source = os.path.join(
-                base_dir,
-                test_configuration.get("util_script"),
-            )
+        util_script = integration_test_fixture.util_script
+        if util_script:
+            script_source = os.path.join(base_dir, util_script)
             os.makedirs(os.path.join(tmp_path, "tests/"))
             util_script_path = os.path.join(tmp_path, "tests/test_utils.py")
             shutil.copyfile(script_source, util_script_path)
 
-        # Check initial state
-
-        # Execute test
-        res = subprocess.run(["python", script_path], capture_output=True)
-        # Check final state
-        expected_stderrs = test_configuration.get("expected_stderrs")
-        expected_stdouts = test_configuration.get("expected_stdouts")
-        expected_failure = test_configuration.get("expected_failure")
-        outs = res.stdout.decode("utf-8")
-        errs = res.stderr.decode("utf-8")
-        print(outs)
-        print(errs)
-
-        if expected_stderrs:
-            assert expected_stderrs == errs
-
-        if expected_stdouts:
-            assert expected_stdouts == outs
-
-        if expected_failure:
-            assert res.returncode != 0
-        else:
-            assert res.returncode == 0
-    except:
-        raise
+        # Run script as module, using python's importlib machinery (https://docs.python.org/3/library/importlib.htm)
+        loader = importlib.machinery.SourceFileLoader("test_script_module", script_path)
+        spec = importlib.util.spec_from_loader("test_script_module", loader)
+        test_script_module = importlib.util.module_from_spec(spec)
+        loader.exec_module(test_script_module)
+    except Exception as e:
+        logger.error(str(e))
```

Comment:
This is truly wonderful stuff! Very much appreciate the capturing of the Exception.